### ISSUE TITLE
Smart Plug Devices not showing in area power overview

### ISSUE THREAD

**Issue by TheSerapher** [tschamm/boschshc-hass#175]
Hey there,

Since March 24th I have issues with my Smart Plug Compact showing properly when assigned to an area. I can see that Powecalc properly added its entities for the device power/energy to the area but my actual power/energy readings from the Bosch plug are missing.

I have tried to change their areas and re-assign them to the old one but I can't get them to show up as an areas power/energy reading. It worked prior to the 24th (when I did updates).

Is there a migration step missing that I failed to do?

![Screenshot 2024-03-30 at 15 01 33](https://github.com/tschamm/boschshc-hass/assets/2376835/d2ab0dda-d094-4aa0-ba4b-119d7e1475d2)
![Screenshot 2024-03-30 at 15 01 19](https://github.com/tschamm/boschshc-hass/assets/2376835/0527b1e4-acb2-4dee-8be1-ce8be481be4f)

**Comment by TheSerapher** [comment]
After checking the energy dashboard and adding each smart plug individually as a device and grid consumption entity the power calculations are correct again - it seems strange that they are not properly used when added to an area (and visible in that area as a device).

**Comment by TheSerapher** [comment]
Closing this since it seems it *might* not be related to the integration itself but rather how Powercalc handled foreign power entities.

Powercalc created metered areas in Home Assistant to track power usage and energy for powercalc devices. It has the option to include devices not from Powercalc which seemed to have worked in the past. With one of the recent upgrades I did those devices were not added to those powerful entities anymore and data wasn’t logged.

I manually added each smart plug into each Powercalc entity as additional power devices and dashboards (including energy) all updated.

Leaving this comment in case someone figures out what happened or if they run into the same issue.
